src/ssl_sess.c

```diff
@@ -1120,7 +1120,9 @@ static int TlsSessionCacheGetAndLock(const byte *id,
 #else
         s = &sessRow->Sessions[idx];
 #endif
-        if (s && XMEMCMP(s->sessionID, id, ID_LEN) == 0 && s->side == side) {
+        /* match session ID value and length */
+        if (s && s->sessionIDSz == ID_LEN && s->side == side &&
```

Comment:
@julek-wolfssl , please review this change. I spent over 8 hours tracking down this long standing bug. Occasionally this would be called with a session id of 0's and return an old releases session and messing up the ex_data. The only solution I could find was to also check `sessionIDSz` since it is set to 0 when the session is released.

Reply:
I suppose the same improvement can be made to `AddSessionToCache` in the `Session already exists. Overwriting.` section. This is a major oversight probably stemming all the way back from when session ID's were the only way to renegotiate. Thanks for finding this.